src/components/FilePreviewer.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import FileIcon from './FileIcon';
+import LoadingIndicator from './LoadingIndicator';
+import type { FileUpload } from '../types';
+
+type Props = {
+  uploads?: FileUpload[];
+  handleRemove?: (id: string) => any;
+  handleRetry?: (id: string) => any;
+  handleFiles?: (files: File[]) => any;
+};
+
+/**
+ * Component that displays files which are being uploaded
+ */
+const FilePreviewer: React.FC<Props> = ({
```

Comment:
just to be safe we should export the component types and rename to FilePreviewerProps etc.

Reply:
Should they also be exported from the index? I think we don't (need to) import these types anywhere, so I'd personally say no, what do you think?

Reply:
yeah we'd typically want to export all component types just in case a TS end user needs to access the type for any reason, so it's less for our sake for more for people using the library. again it's not super critical in this lib since the components are pretty simple, but in the other TS libraries we've been erring on the side of exporting all components types just to give people the option